projects/plugins/jetpack/modules/shortcodes/spotify.php

```diff
@@ -48,6 +48,26 @@ function jetpack_spotify_shortcode( $atts = array(), $content = '' ) {
 	// Spotify accepts both URLs and their Spotify ID format, so let them sort it out and validate.
 	$embed_url = add_query_arg( 'uri', rawurlencode( $id ), 'https://embed.spotify.com/' );
 
+	// If the shortcode is displayed in a WPCOM notification, display a simple link only.
+	// When the shortcode is displayed in the WPCOM Reader, use iframe instead.
+	if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+		require_once WP_CONTENT_DIR . '/lib/display-context.php';
+		$context = A8C\Display_Context\get_current_context();
+		if ( A8C\Display_Context\NOTIFICATIONS === $context ) {
+			return sprintf(
+				'<a href="%1$s" target="_blank" rel="noopener noreferrer">%1$s</a>',
+				esc_url( $id )
+			);
+		} elseif ( A8C\Display_Context\READER === $context ) {
+			return sprintf(
+				'<iframe src="%1$s" height="%2$s" width="%3$s"></iframe>',
+				esc_url( $embed_url ),
+				esc_attr( $atts['height'] ),
+				esc_attr( $atts['width'] )
+			);
```

Comment:
Is there a specific reason why we need that fallback, given that we render an iFrame below? Are the inline styles added to the iFrame below causing issues in the Reader?

Reply:
> Is there a specific reason why we need that fallback, given that we render an iFrame below? Are the inline styles added to the iFrame below causing issues in the Reader?

Thanks for the comment Jeremy. Yes, that's right the fallback is there to fix the "black box" below the player due to stripped down inline styles in Reader.

Page itself:
![Markup on 2021-09-10 at 15:45:26](https://user-images.githubusercontent.com/25105483/132863300-8e4b21da-3f06-4a1d-b44e-bb6f569cdce6.png)

Reader (without the fix):
![Markup on 2021-09-10 at 15:46:09](https://user-images.githubusercontent.com/25105483/132863328-bafd93e3-cd52-49a1-9da9-80608786703b.png)

That is the reason I suggested to use the `height` and `width` attributes directly.

I should have added clarification on that in the PR.